connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailCloudEventDataExtractor.java

```diff
@@ -0,0 +1,46 @@
+package com.redhat.cloud.notifications.connector.email;
+
+import com.redhat.cloud.notifications.connector.CloudEventDataExtractor;
+import com.redhat.cloud.notifications.connector.email.constants.Properties;
+import com.redhat.cloud.notifications.connector.email.decoder.ActionDecoder;
+import com.redhat.cloud.notifications.connector.email.model.User;
+import com.redhat.cloud.notifications.connector.email.model.settings.RecipientSettings;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.camel.Exchange;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashSet;
+import java.util.List;
+
+@ApplicationScoped
+public class EmailCloudEventDataExtractor extends CloudEventDataExtractor {
+    /**
+     * Extracts the relevant information for the email connector from the
+     * received message from the engine.
+     * @param exchange the incoming exchange from the engine.
+     * @param cloudEventData the received Cloud Event.
+     */
+    @Override
+    public void extract(final Exchange exchange, final JsonObject cloudEventData) {
+        final JsonArray settings = cloudEventData.getJsonArray("recipient_settings");
+
+        final List<RecipientSettings> recipientSettings = settings
+            .stream()
+            .map(JsonObject.class::cast)
+            .map(jsonSetting -> jsonSetting.mapTo(RecipientSettings.class))
+            .toList();
+
+        final JsonArray subs = cloudEventData.getJsonArray("subscribers_uuids");
```

Comment:
:information_source: Our code/DB contains misnamed fields/methods. The actual data that we use for subscriptions is the `username` field from the Red Hat identity header.

Anyway, we could use a simpler name here:
```suggestion
        final JsonArray subs = cloudEventData.getJsonArray("subscribers");
```

Reply:
Changed!